### ISSUE TITLE
bootstrap-salt.sh (2020.10.20) fails on CentOS 7: ERROR: Failed to run install_centos_git()!!!

### ISSUE THREAD

**Issue by juliank** [saltstack/salt-bootstrap#1527]
### Description of Issue/Question
I've downloaded the latest version of the bootstrap script to a new and "clean" CentOS 7 server. Running the script eventually fails with the error message `ERROR: Failed to run install_centos_git()!!!`. I have also tried using the latest version of the script from the develop branch, but the issue is the same.

### Setup
The server is a new and clean (fresh from our operations department) server running CentOS Linux release 7.7.1908 (Core). No other installations or configurations has been done on the server prior to attempting to install the salt minion.

### Steps to Reproduce Issue
```
curl -o bootstrap-salt.sh -L https://bootstrap.saltstack.com
sudo sh bootstrap-salt.sh -P -A 10.254.20.1 git v3002.2
```
After going through lots of downloads and installations, it finally fails with this error:

```
 *  INFO: Installing salt using python
 *  INFO: Downloading Salt Dependencies from PyPi
Processing /tmp/git/salt
  Link is a directory, ignoring download_dir
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-DZ_NvE-build/setup.py", line 178, in <module>
        exec(compile(open(SALT_VERSION).read(), SALT_VERSION, "exec"))
      File "/tmp/pip-DZ_NvE-build/salt/version.py", line 789
        yield from info
                 ^
    SyntaxError: invalid syntax

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-DZ_NvE-build/
You are using pip version 8.1.2, however version 20.3.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Failed to download salt dependencies
 *  INFO: Installing Downloaded Salt Dependencies
Invalid requirement: '/tmp/git/deps/*'
It looks like a path. Does it exist ?
You are using pip version 8.1.2, however version 20.3.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
 * ERROR: Failed to run install_centos_git()!!!
```

See [complete log here](https://gist.github.com/juliank/9a6c69de47a2568dbc1a2910076cd57f).

I have also tried using the latest version of the bootstrap script from the develop branch, but this yields the same error. This log (including debug output) [can be found here](https://gist.github.com/juliank/8fd7c29c05b2026873451f620d847285).

On another (also clean) CentOS 7 server I also tried upgrading pip (as suggested by the bootstrap script: `pip install --upgrade pip`), but the same error is still present. (Along with a lot of warnings about Python 2.7 being deprecated).

I have successfully installed the salt master (using the same bootstrap script) on a CentOS 8 server in the same environment, so there should hopefully not be any "surrounding" issues causing this.

### Versions and Systems
`bootstrap-salt.sh -v`: Version 2020.10.20
Server (VM): CentOS Linux release 7.7.1908 (Core)

**Comment by bryceml** [comment]
versions greater than 3001 and newer require `-x python3` specified unless the os only ever was supported on python3.

for example ubuntu 20.04 only was ever supported on python3, so the `-x python3` is not needed.  Centos 7 was supported with python2, so `-x python3` is needed.

it doesn't hurt to just throw `-x python3` on anything 3001 or newer.

Once py2 is no longer supported at all, I think it would make sense to change the default to python3 for all oses.  We probably need some discussion on that since it would break people using unsupported versions of salt.

**Comment by juliank** [comment]
Thank you @bryceml, `-x python3` did the trick.

I noticed the `-x` option, but misread the documentation. I actually tried running with `-x python2.7` - my line of thought being that since Python 2.7 is the version present on the OS, I might have to force the bootstrap script to use this old and deprecated version. If this misunderstanding is on me, or the documentation might be in need of improvement, I'll leave for others to decide.
